ci/ray_ci/tester.py

```diff
@@ -349,9 +364,35 @@ def _get_test_targets(
     )
     flaky_tests = set(_get_flaky_test_targets(team, operating_system, yaml_dir))
 
+    # normal case, we want to exclude flaky tests from the list of targets provided
+    # by users
+    final_targets = test_targets.difference(flaky_tests)
+
     if get_flaky_tests:
-        return list(flaky_tests.intersection(test_targets))
-    return list(test_targets.difference(flaky_tests))
+        # run flaky test cases, so we include flaky tests in the list of targets
```

Comment:
maybe this should be:

````
if get_flaky_tests:
    final_targets = test_targets.intersection(flaky_tests)
elif
    final_targets = test_targets.difference(flaky_tests)
````

I am a bit confused by the overwriting of `final_targets` here.

Reply:
updated